core/trino-main/src/main/java/io/trino/execution/QueryExecution.java

```diff
@@ -99,22 +99,27 @@ interface QueryExecutionFactory<T extends QueryExecution>
      */
     class QueryOutputInfo
     {
-        private final List<String> columnNames;
+        private final List<ColumnInfo> columnInfos;
         private final List<Type> columnTypes;
         private final Queue<ExchangeInput> inputsQueue;
         private final boolean noMoreInputs;
 
-        public QueryOutputInfo(List<String> columnNames, List<Type> columnTypes, Queue<ExchangeInput> inputsQueue, boolean noMoreInputs)
+        public QueryOutputInfo(List<ColumnInfo> columnInfos, List<Type> columnTypes, Queue<ExchangeInput> inputsQueue, boolean noMoreInputs)
```

Comment:
This now looks much better than 6 separate lists, thank you!

Should Type be part of ColumnInfo?